test/blackbox-tests/test-cases/oxcaml/implements-parameter.t

```diff
@@ -0,0 +1,135 @@
+This test ensures the `implements` field is working in the context of
+parameterized libraries.
+
+  $ . ./helpers.sh
+
+We create two parameters, one public, the other one local.
+
+  $ init_project
+
+  $ make_dir_with_dune "foo" <<EOF
+  > (library_parameter
+  >   (public_name foo))
+  > EOF
+  $ make_dummy_intf "foo" "foo"
+  $ cat >> dune-project << EOF
+  > (package (name foo))
+  > EOF
+
+  $ make_dir_with_dune "bar" <<EOF
+  > (library_parameter
+  >   (name bar))
+  > EOF
+  $ make_dummy_intf "bar" "bar"
+
+  $ dune build $(target_cmi "bar")
+  $ dune build $(target_cmi "foo")
+
+  $ make_dir_with_dune "foo_lib" <<EOF
+  > (library
+  >  (name foo_lib))
+  > EOF
+  $ cat > "foo_lib/foo_lib.ml" <<EOF
+  > let x = 42
+  > EOF
+  $ dune build
+
+
+Trying to implement a non-parameter and non-virtual library should fail with the
+correct error message.
+
+  $ rm -rf _build
+  $ make_dir_with_dune "foo_impl" <<EOF
+  > (library
+  >  (name foo_impl)
+  >  (implements missing_foo))
+  > EOF
+  $ dune build
+  File "foo_impl/dune", line 3, characters 13-24:
+  3 |  (implements missing_foo))
+                   ^^^^^^^^^^^
+  Error: Library "missing_foo" not found.
+  -> required by alias default
+  [1]
+
+We implement the parameter using a library calling a wrong parameter name.
+
+  $ rm -rf _build foo_impl
+  $ make_dir_with_dune "foo_impl" <<EOF
+  > (library
+  >  (name foo_impl)
+  >  (implements missing_foo))
+  > EOF
+  $ dune build
+  File "foo_impl/dune", line 3, characters 13-24:
+  3 |  (implements missing_foo))
+                   ^^^^^^^^^^^
+  Error: Library "missing_foo" not found.
+  -> required by alias default
+  [1]
```

Comment:
These two tests appear to be checking the exact same operations with the same inputs, and this doesn't make sense to me. @maiste has tried to explain this at https://github.com/ocaml/dune/pull/12002#discussion_r2300153552 , but I still do not how running the exact same test with two different descriptions can be useful :sweat_smile: .

I suspect the test needs to be fixed.